lib/vsc/mympirun/scoop/myscoop.py

```diff
@@ -337,27 +347,56 @@ def scoop_run(self):
-                          self.scoop_profile,
-                          self.scoop_pythonpath[0],
-                          # custom
-                          self.scoop_freeorigin,
-                          self.scoop_processcontrol,
-                          self.scoop_affinity,
-                          vars_to_pass,
-                          ]
+        if LooseVersion(SCOOP_VERSION) >= LooseVersion('0.7'):
+            scoop_app_args = [
+                [(nodename, len(list(group))) for nodename, group in itertools.groupby(self.scoop_hosts)],  # hosts
+                self.scoop_size,  # n
+                1,  # b (total number of brokers to spawn on the hosts, one by default)
+                self.scoop_verbose,  # verbose
+                [self.scoop_python],  # python_executable
+                self.scoop_broker,  # externalHostName
+                self.scoop_executable,  # executable
+                self.scoop_args,  # arguments
+                self.scoop_tunnel,  # tunnel
+                self.scoop_path,  # path
+                self.scoop_debug,  # debug
+                self.scoop_nice,  # nice
+                "other",  # env; TODO check utils.getEnv(),
+                self.scoop_profile,  # profile
+                self.scoop_pythonpath[0],  # pythonPath
+                [None],  # prolog (path to prolog script, default is None)
+                'ZMQ',  # backend (ZMQ or TCP, default: ZMQ)
+                #False, # rsh (default)  added in future version?
+                # custom
+                self.scoop_freeorigin,
+                self.scoop_processcontrol,
+                self.scoop_affinity,
+                vars_to_pass,
+            ]
+        else:
+            scoop_app_args = [
+                [(nodename, len(list(group))) for nodename, group in itertools.groupby(self.scoop_hosts)],  # hosts
+                self.scoop_size,  # n
+                self.scoop_verbose,  # verbose
+                [self.scoop_python],  # python_executable
+                self.scoop_broker,  # brokerHostName
+                self.scoop_executable,  # executable
+                self.scoop_args,  # arguments
+                self.scoop_tunnel,  # tunnel
+                None,  # log; TODO args.log, deal with fancylogger later
+                self.scoop_path,  # path
+                self.scoop_debug,  # debug
+                self.scoop_nice,  # nice
+                "other",  # env; TODO check utils.getEnv(),
+                self.scoop_profile,  # profile
+                self.scoop_pythonpath[0],  # pythonPath
+                # custom
+                self.scoop_freeorigin,
+                self.scoop_processcontrol,
+                self.scoop_affinity,
+                vars_to_pass,
+            ]
+
         self.log.debug("scoop_run: scoop_app class %s args %s" % (self.SCOOP_APP.__name__, scoop_app_args))
 
         scoop_app = self.SCOOP_APP(*scoop_app_args)
```

Comment:
this is bound to break again in the future, can't we pass named arguments? and a nice error if it seems like the api changed?